src/replica_ctrl.py

```diff
@@ -155,31 +128,74 @@ def create_mongo_config(tasks_ips, replicaset_name, mongo_port):
     return config
 
 
-def update_config(primary_ip, current_ips, new_ips, replicaset_name, mongo_port):
+def gather_configured_members_ips(mongo_tasks_ips, mongo_port):
+    current_ips = set()
+    for t in mongo_tasks_ips:
+        mc = pm.MongoClient(t, mongo_port)
+        try:
+            config = mc.admin.command("replSetGetConfig")['config']
+            for m in config['members']:
+                current_ips.add(m['host'].split(":")[0])
```

Comment:
if the for loop complete successfully, you should return. this avoid that multiple "dirty" configuration from different nodes are loaded. i think already made some changes in that respect in my branch